src/constructs/autoscaling/asg.ts

```diff
@@ -11,6 +11,8 @@ import { GuAmiParameter, GuInstanceTypeParameter } from "../core";
 // https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys
 export interface GuAutoScalingGroupProps
   extends Omit<AutoScalingGroupProps, "imageId" | "osType" | "machineImage" | "instanceType" | "userData"> {
+  instanceType?: InstanceType;
+  imageId?: string;
```

Comment:
This would require the prop to be a parameter so we can cater for stacks with multiple ASGs but enforce using parameters for AMIs.

```suggestion
  imageId?: GuAMIParameter;
```

Reply:
That's a nice idea 👍